### ISSUE TITLE
/api/v0/pubsub/pub should use body for payload

### ISSUE THREAD

**Issue by hugomrdias** [ipfs/http-api-docs#28]
https://docs.ipfs.io/reference/api/http/#api-v0-pubsub-pub states that payload should go in a querystring param called `arg` while i think this somehow works with some serialisation it should say to use the request body instead, just like the dht/put https://docs.ipfs.io/reference/api/http/#api-v0-dht-put

We are updating js-ipfs-http-client to use body instead of querystring in this PR https://github.com/ipfs/js-ipfs/pull/3013.

**Comment by lidel** [comment]
Indeed. Things changed since 2020 but js-ipfs-http-client seems to send data in body instead of second `arg` in URL:   https://github.com/ipfs/js-ipfs/blob/f11220e00a12afed5ebbbd8b4c5134595aea735d/packages/ipfs-http-client/src/pubsub/publish.js#L32

Need to double check if/how this impacts https://github.com/ipfs/go-ipfs/pull/8183
